src/Messenger/Middleware/NewRelicMiddleware.php

```diff
@@ -49,6 +51,19 @@ public function handle(Envelope $envelope, StackInterface $stack): Envelope
 
         try {
             return $stack->next()->handle($envelope, $stack);
+        } catch (\Throwable $exception) {
+            $wrappedExceptions = $exception instanceof WrappedExceptionsInterface ? $exception->getWrappedExceptions() : [$exception];
+            foreach ($wrappedExceptions as $wrappedException) {
+                foreach ($this->excludedExceptions as $excludedException) {
+                    if ($wrappedException instanceof $excludedException) {
+                        throw $exception;
+                    }
+                }
+            }
+            // Mark current transaction as error, but ensure NR groups it by OG exception and not wrapped one when possible
+            $this->interactor->noticeThrowable(1 === \count($wrappedExceptions) ? reset($wrappedExceptions) : $exception);
```

Comment:
@ostrolucky I'm wondering if this is the optimal location for these checks. Would it be more appropriate to relocate them to `ExceptionListener`? This way, we could centralize the logic of all application exceptions in one location. What are your thoughts on this?

Reply:
I can put it to listener if you want. Personally I wasn't comfortable with that, because unlike listener, middleware has advantage of having access to a stack. I think this might help us with current issue this bundle has: Let's say message handler is doing this:

```php
 public function __invoke(Foo $message): void
    {
        $this->messageBus->dispatch(new Bar());
        throw new \Exception('foo');
    }
```
Can you guess transaction name of the failure? Well, it's `Bar`. I just tried. Despite exception actually being thrown in handler for `Foo`. This is something we can perhaps do something about at some point in future, but I feel we might have better chance of solving this in middleware.

Btw transaction starting/stopping could have also been done in listener instead of having a messenger middleware.